src/soc/intel/common/block/oc_wdt/acpi.c

```diff
@@ -0,0 +1,156 @@
+
+#define OC_WDT_MIN_VALUE	0
+#define OC_WDT_MAX_VALUE	1023
+#define OC_WDT_PERIOD_MS	1000
+
+static bool fill_wdat_timeout_entry(acpi_wdat_entry_t *entry)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = ACPI_WDAT_SET_COUNTDOWN;
+	entry->instruction = ACPI_WDAT_WRITE_COUNTDOWN | ACPI_WDAT_PRESERVE_REGISTER;
+	entry->mask = PCH_OC_WDT_CTL_TOV_MASK;
+	entry->register_region.space_id = ACPI_ADDRESS_SPACE_IO;
+	entry->register_region.addrl = PCH_OC_WDT_CTL;
+	entry->register_region.access_size = ACPI_WDAT_ACCESS_SIZE_DWORD;
+
+	return true;
+}
+
+static bool fill_wdat_boot_status_entry(acpi_wdat_entry_t *entry, uint8_t action,
+					uint8_t instruction, uint32_t value)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = action;
+	entry->instruction = instruction;
+	entry->value = value;
+	entry->mask = PCH_OC_WDT_CTL_SCRATCH_MASK;
+	entry->register_region.space_id = ACPI_ADDRESS_SPACE_IO;
+	entry->register_region.addrl = PCH_OC_WDT_CTL;
+	entry->register_region.access_size = ACPI_WDAT_ACCESS_SIZE_DWORD;
+
+	return true;
+}
+
+static bool fill_wdat_run_state_entry(acpi_wdat_entry_t *entry, uint8_t action,
+				       uint8_t instruction, uint32_t value)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = action;
+	entry->instruction = instruction;
+	entry->value = value;
+	entry->mask = PCH_OC_WDT_CTL_EN;
+	entry->register_region.space_id = ACPI_ADDRESS_SPACE_IO;
+	entry->register_region.addrl = PCH_OC_WDT_CTL;
+	entry->register_region.access_size = ACPI_WDAT_ACCESS_SIZE_DWORD;
+
+	return true;
+}
+
+static bool fill_wdat_ping_entry(acpi_wdat_entry_t *entry)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = ACPI_WDAT_RESET;
+	entry->instruction = ACPI_WDAT_WRITE_VALUE | ACPI_WDAT_PRESERVE_REGISTER;
+	entry->value = PCH_OC_WDT_CTL_RLD;
+	entry->mask = PCH_OC_WDT_CTL_RLD;
+	entry->register_region.space_id = ACPI_ADDRESS_SPACE_IO;
+	entry->register_region.addrl = PCH_OC_WDT_CTL;
+	entry->register_region.access_size = ACPI_WDAT_ACCESS_SIZE_DWORD;
+
+	return true;
+}
+
+unsigned long acpi_soc_fill_wdat(acpi_wdat_t *wdat, unsigned long current)
+{
+	if (!wdat)
+		return current;
+
+	wdat->pci_segment = 0xff;
+	wdat->pci_bus = 0xff;
+	wdat->pci_device = 0xff;
+	wdat->pci_function = 0xff;
+
+	wdat->timer_period = OC_WDT_PERIOD_MS;
+	wdat->min_count = OC_WDT_MIN_VALUE;
+	wdat->max_count = OC_WDT_MAX_VALUE;
+	wdat->flags = ACPI_WDAT_FLAG_ENABLED;
+	wdat->entries = 0;
+
+	acpi_wdat_entry_t *entry = (acpi_wdat_entry_t *)current;
+
+	/* Write countdown */
+	if (!fill_wdat_timeout_entry(entry))
+		goto out_err;
+
+	entry++;
+
+	/* Get boot status */
+	if (!fill_wdat_boot_status_entry(entry, ACPI_WDAT_GET_STATUS,
+					 ACPI_WDAT_READ_VALUE, 1))
+		goto out_err;
+
+	entry++;
+
+	/* Set boot status */
+	if (!fill_wdat_boot_status_entry(entry, ACPI_WDAT_SET_STATUS,
+					 ACPI_WDAT_WRITE_VALUE | ACPI_WDAT_PRESERVE_REGISTER,
+					 0))
```

Comment:
This should also be handled in a way that the Status bits are not cleared if they don't need to be

Reply:
https://github.com/Dasharo/coreboot/pull/472/commits/f99cf95497da3151fe21dc6806c23af9b32e5129